metaflow/plugins/test_unbounded_foreach_decorator.py

```diff
@@ -97,7 +133,7 @@ def control_task_step_func(self, flow, graph, retry_count):
             task_id = "%s-%d" % (control_task_id.replace("control-", "test-ubf-"), i)
             pathspec = "%s/%s/%s" % (run_id, step_name, task_id)
             mapper_tasks.append(to_unicode(pathspec))
-            input_paths = "%s/%s/%s" % (run_id, split_step_name, split_task_id)
+            input_paths = ",".join(self.input_paths)
```

Comment:
ok, so this removes the need to name the control task a particular way yes? I see that it now also has possibly more than one? What is the use case here?

Reply:
instead of constructing the input_paths dynamically, we are just passing them as passed down by the lifecycle sine the task_pre_step gets the input paths. More than this being usecase-specific, this is a simplification of the code since this is a more "natural" way of retrieving the input_paths (instead of dynamically constructing them using the control task id). 

Control tasks will have special metadata that will help mark if the task is a control task. The worker tasks will generate their own task-ids based on task-id passed by the control task.